src/Source/VectorTilesSource.js

```diff
@@ -88,7 +57,74 @@ class VectorTilesSource extends TMSSource {
             }));
         }
 
-        this.whenReady = Promise.all(promises);
+        this.whenReady = Promise.all(promises).then((style) => {
+            style = style[0];
+            const s = Object.keys(style.sources)[0];
+            const os = style.sources[s];
+            let order = 0;
+
+            style.layers.forEach((layer) => {
+                layer.sourceUid = this.uid;
+                if (layer.type === 'background') {
+                    this.backgroundLayer = layer;
+                } else if (ffilter(layer)) {
+                    // TODO: add support for expressions
+                    // https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions
+                    const stops = [];
+                    stops.push(layer.minzoom == undefined ? 2 : layer.minzoom);
+                    checkStopsValues(layer.layout, stops);
+                    checkStopsValues(layer.paint, stops);
+                    stops.push(layer.maxzoom == undefined ? 24 : layer.maxzoom);
+                    stops.sort((a, b) => (a - b));
+
+                    this.styles[layer.id] = [];
+                    for (let i = 0; i < stops.length - 1; i++) {
+                        if (stops[i] == stops[i + 1]) { continue; }
+                        const style = new Style();
+                        style.minzoom = stops[i];
+                        style.maxzoom = stops[i + 1];
+                        style.setFromVectorTileLayer(layer, this.sprites, this.symbolToCircle);
+                        this.styles[layer.id].push(style);
+                    }
+
+                    if (!this.filter[layer['source-layer']]) {
+                        this.filter[layer['source-layer']] = [];
+                    }
+
+                    this.filter[layer['source-layer']].push({
+                        id: layer.id,
+                        order: order++,
```

Comment:
Good catch, I'll see what I can do

Reply:
It seems that we can't escape a `sort`, as the loop is built around the received features, and looping around the whole `filter` wouldn't be a good idea too.

Reply:
Could you sort `this.filter` after the end of its construction?

Reply:
Impossible, it is an object, and I made it that way because it is much faster than an array to read (`obj[key]` vs `arr.indexOf(key)` for example). But as I said, we can't escape the `sort` as we don't know the order features in the `.pbf/.mvt` are.

Reply:
Ok, I didn't realize it was an object. we'll see later to avoid being sorted .